pkg/applier/applier.go

```diff
@@ -163,25 +161,29 @@ func NewSupervisor(cs *ClientSet, scope declared.Scope, syncName string, reconci
 // objects in a single namespace.
 func NewNamespaceSupervisor(cs *ClientSet, namespace declared.Scope, syncName string, reconcileTimeout time.Duration) (Supervisor, error) {
 	syncKind := configsync.RepoSyncKind
-	invObj := newInventoryUnstructured(syncKind, syncName, string(namespace), cs.StatusMode)
+	syncNamespace := namespace.SyncNamespace()
+	invObj := newInventoryUnstructured(syncName, syncNamespace)
 	// If the ResourceGroup object exists, annotate the status mode on the
 	// existing object.
 	if err := annotateStatusMode(context.TODO(), cs.Client, invObj, cs.StatusMode); err != nil {
```

Comment:
Do we need to annotate the status mode in the constructor? Can we move this to the converter?

Reply:
It's also set in the converter. I didn't remove this because I assumed there may be good reason to set this annotation before the applier runs, e.g. to unwedge a ResourceGroup whose status is too large

Reply:
I don't think we need it here. 

- Applier.Run will create a new one for us with the right annotation the first time. 
- It will never be too big on the cluster, because it would get denied before writing to etcd.
- NewNamespaceSupervisor only gets called once when the reconciler initializes. So it can only really "fix" anything very occasionally, and even so, the applier will run shortly after initialization, doing the same thing.

Plus, removing it reduces uses of  newInventoryUnstructured, which seems like something we shouldn't have outside of the converter code or tests.

Reply:
If the annotation didn't previously exist, it could be added here, and that would trigger the RGC to remove the status, but that seems unnecessary, because InvAddTask would do the same thing a few seconds later at most.

Reply:
Fair points, I'm on board to remove it.

As a side note: It occurred to me that if a ResourceGroup transitions from statusEnabled -> statusDisabled, the applier won't clear the status (since it skips UpdateStatus). Rather it will set the statusDisabled annotation on the ResourceGroup and then afterwards the resource-group-controller will clear the status. This is probably fine although it may be preferable for the applier to actually clear the status, I'm unsure.

Reply:
All that being said, there's an edge case where if a user has too many managed objects they wouldn't be able to disable the status. This is because the applier would be trying to set the annotation on the ResourceGroup with the same update that it's adding objects to the inventory (exceeding the size limit). So the applier would not be able to set the annotation until the user removed objects from the source repository.

This seems less user friendly than the alternative of setting the annotation with this separate initial update

Reply:
Can you fix this in a separate PR? Doesn't need to block.

It shouldn't be making API calls in a constructor. But it also doesn't need to run it before every apply. It just needs to run once, perhaps wrapped with RetryOnConflict (Get+Update), or just using a patch.

Reply:
https://github.com/GoogleContainerTools/kpt-config-sync/pull/1619